### ISSUE TITLE
Model Behaviors

### ISSUE THREAD

**Issue by Skullbock** [akeeba/fof#32]
During our Skype discussion, we agreed that it's necessary to split the code present in the current MVC classes to make it easier to manage and also to be able to "pick and choose" some of the features a model could have.
### Models

I was working on the integration of the filters i already implemented for some of our extensions at zoolanders.com like:
- partial text search
- exact text search
- any word text search
- all words text seach
- exact number search
- intervals (>, <, >=, <=, <>, etc)
- dates (>, <, >=, <=, <>, +10 days, etc)

This kind of addition lead to a lot of extra code in the model, therefore i seriusly started search for a good solution
### Behaviors

I was very taken by the way nooku managed this by the use of behaviors, but i didn't like how they used mixins to achieve it.
Therefore i started searching for alternatives, and i found that (strangely) **CakePHP** does this in a clever way (IMHO), which actually suits FOF very well.

In CakePHP, you can add behaviors to the models, separating features very easily. This gets pulled into the main model by the use of the (already present in FOF) before / after events.

Basically, you can create a model with a list of behaviors (and also attach / detach them later on). 
These behaviors can add and remove features by implementing the same onBeforexxx and onAfterxxx that the current FOF models have.

Check their repo for some code examples:
https://github.com/cakephp/cakephp/tree/master/lib/Cake/Model

Can i take this is a good way to start?
